### DISCUSSION TITLE
Unable to dynamically set env vars

### DISCUSSION THREAD

**Discussion by mihey278** [vercel/next.js#35821] [Help]
My goal is to set an env variable that can be evaluated at time of dev/prod server start. I followed the guide in here:

https://nextjs.org/docs/basic-features/environment-variables

but it seems the interpreter interprets everything literally (as opposed to evaluate bash commands, as would be expected). Is there a way to enable that? The current setup is limiting in the sense that all env vars need to be known upfront (i.e. I cannot set an env var to the my local dev server's ip).

More details here: https://stackoverflow.com/questions/71573508/how-to-evaluate-command-in-next-js-env-file

**Comment by icyJoseph** [comment]
Hi,

What version of next.js are you using, and why would you expect it to evaluate bash commands? 

Under the hood, I think this module is used https://github.com/motdotla/dotenv-expand, to do things like this:

```shell
# .env
HOSTNAME=localhost
PORT=8080
HOST=http://$HOSTNAME:$PORT
```

I am gonna investigate a bit if `process` has the variable you are looking for.

I wonder if your use case could be fixed with runtime config, https://nextjs.org/docs/api-reference/next.config.js/runtime-configuration, like so:

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  serverRuntimeConfig: {
    // Will only be available on the server side
    secondSecret: process.env.SECOND_SECRET, // Pass through env variables
  },
};

module.exports = nextConfig;
```

And then after you build the application, you run it with `SECOND_SECRET hello yarn start` for example, and then wherever you use `secondSecret` it will be hello. 

For example here it'd log the value:

```ts
import getConfig from "next/config";

export default function handler(req, res) {
  const { serverRuntimeConfig } = getConfig();
  console.log({ serverRuntimeConfig });
  res.status(200).json({ name: "John Doe" });
}
```

Thought this has limitations when it comes to code that arrives to the client.
